core/modules/qdisp/QueryRequest.cc

```diff
@@ -59,71 +59,45 @@ inline void unprovisionSession(XrdSsiSession* session) {
 }
 
 ////////////////////////////////////////////////////////////////////////
-// QueryRequest::Canceller
-////////////////////////////////////////////////////////////////////////
-class QueryRequest::Canceller : public util::VoidCallable<void> {
-public:
-    Canceller(QueryRequest* qr) : _queryRequest(qr) {}
-    virtual ~Canceller() { delete _queryRequest; }
-    virtual void operator()() {
-        if (_queryRequest != NULL) {
-            _queryRequest->cancel();
-        }
-    }
-private:
-    QueryRequest* _queryRequest;
-};
-////////////////////////////////////////////////////////////////////////
 // QueryRequest
 ////////////////////////////////////////////////////////////////////////
-QueryRequest::QueryRequest(
-    XrdSsiSession* session,
-    std::string const& payload,
-    std::shared_ptr<ResponseRequester> const requester,
-    std::shared_ptr<util::UnaryCallable<void, bool> > const finishFunc,
-    std::shared_ptr<util::VoidCallable<void> > const retryFunc,
-    JobStatus& status)
-    : _session(session),
-      _payload(payload),
-      _requester(requester),
-      _finishFunc(finishFunc),
-      _retryFunc(retryFunc),
-      _status(status),
-      _finishStatus(QueryRequest::ACTIVE) {
-    LOGF_INFO("New QueryRequest with payload(%1%)" % payload.size());
-    _registerSelfDestruct();
+QueryRequest::QueryRequest( XrdSsiSession* session, std::shared_ptr<JobQuery> const& jobQuery) :
+    _session(session), _jobQuery(jobQuery), _jobDesc(_jobQuery->getDescription()) {
+    LOGF_INFO("New QueryRequest with payload(%1%)" % _jobDesc.payload().size());
 }
 
 QueryRequest::~QueryRequest() {
+    LOGF_DEBUG("~QueryRequest");
     unprovisionSession(_session);
```

Comment:
unprovisionSession() seems to be called from this place only.  Just put the def inline here?